storage/ndb/rest-server/rest-api-server/internal/feature_store/avro_parser.go

```diff
@@ -0,0 +1,215 @@
+/*
+ * This file is part of the RonDB REST API Server
+ * Copyright (c) 2023 Hopsworks AB
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package feature_store
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/hamba/avro/v2"
+
+	"hopsworks.ai/rdrs/internal/log"
+)
+
+// parser
+func ConvertAvroSchemaToStruct(schema avro.Schema) (reflect.Type, error) {
+	log.Debug(fmt.Sprintf("\n--> ConvertAvroSchemaToStruct Called %v\n", schema))
+	switch schema.Type() {
+	case avro.Record:
```

Comment:
Horrible indentation, is it a must to use such horrible indendation rules?

Reply:
All files are formatted according to the formatting rules.